backend/typescript/rest/creatorRoutes.ts

```diff
@@ -1,11 +1,114 @@
 import { Router } from "express";
-import { creatorDtoValidator } from "../middlewares/validators/creatorValidators";
+import { isAuthorizedByRole } from "../middlewares/auth";
+import CreatorService from "../services/implementations/creatorService";
+import ICreatorService from "../services/interfaces/creatorService";
+import { sendErrorResponse } from "../utilities/errorResponse";
 
 const creatorRouter: Router = Router();
 creatorRouter.get("/");
 
-creatorRouter.get("/", creatorDtoValidator, async (req, res) => {
-  res.status(200).send("hello creator");
-});
+const creatorService: ICreatorService = new CreatorService();
+
+function isOverlap(ageRange: string, ageRangeCmp: string): boolean {
+  const lowerInclusiveBound = parseInt(
+    ageRange.split(",")[0].slice(1) || "",
```

Comment:
Not quite sure what this defaults to or what the `||` is for? For one, if anything goes wrong here i.e `ageRange` is not a string or is malformed, then this will throw an error, so the OR condition will never be used. I think it would be better to just get rid of the OR statement entirely and ensure the string is always well formed